homeassistant/components/deconz/light.py

```diff
@@ -208,6 +223,15 @@ def __init__(self, device: _LightDeviceT, hub: DeconzHub) -> None:
         if device.effect is not None:
             self._attr_supported_features |= LightEntityFeature.EFFECT
             self._attr_effect_list = [EFFECT_COLORLOOP]
+
+            # For lights that report supported effects.
+            if isinstance(self._device, Light):
+                light = cast(Light, device)
+                if light.supported_effects is not None:
+                    self._attr_effect_list = [
+                        el.value for el in light.supported_effects
+                    ]
```

Comment:
```suggestion
            if isinstance(device, Light):
                if device.supported_effects is not None:
                    self._attr_effect_list = device.supported_effects
```
If you use the same object all the time you can probably get away with removing a couple of steps.